operations/iuf/workflows/upgrade_all_products.md

```diff
@@ -1,120 +1,393 @@
     ```
 
-    Once this step has completed:
+1. Inspect the newly-created management NCN and managed node images and CFS configurations to ensure they are correct before continuing with the next steps of the workflow.
+
+Once this step has completed:
+
+- New CFS configurations have been created for management NCNs and managed compute and application (UAN, etc.) nodes
+- New images have been created for management NCNs and managed compute and application (UAN, etc.) nodes
+- New BOS session templates have been created to boot managed compute and application (UAN, etc.) nodes with the new images and CFS configurations
+- Per-stage product hooks have executed for the `update-cfs-config` and `prepare-images` stages
+
+## Backup
+
+This section describes procedures that backup critical state in case it becomes necessary to fall back to previous configurations and software.
+
+TBD
+
+## Management nodes rollout
+
+This section updates the software running on management NCNs.
 
-    - new CFS configurations have been created for management NCNs and managed compute and application (UAN) nodes
-    - new images have been created for management NCNs and managed compute and application (UAN) nodes
-    - new BOS session templates have been created to boot managed compute and application (UAN) nodes with the new images and CFS configurations
-    - per-stage product hooks have executed for the `update-cfs-config` and `prepare-images` stages
+### Update management host firmware (FAS)
 
-1. Inspect the newly-created management NCN and managed node images and CFS configurations to ensure they are correct before rebooting to the new content.
+TBD
 
-1. Invoke `iuf run` with `-r` to execute the [`management-nodes-rollout`](../stages/management_nodes_rollout.md) stage. The `-m`, `--site-vars`, and `--bootprep-config-dir` arguments are the same as the initial invocation of `iuf run`
-   as all steps in this example use the same IUF activity, media, and variables.
+### Execute the IUF `management-nodes-rollout` stage
 
-    (`ncn-m001#`) Execute the `management-nodes-rollout` stage.
+**`NOTE`** Additional arguments are available to control the behavior of the `management-nodes-rollout` stage, for example `--limit-management-rollout` and `-cmrp`. See the
+[`management-nodes-rollout` stage documentation](../stages/management_nodes_rollout.md) for details and adjust the examples below if necessary.
+
+1. Refer to the "Install and Upgrade Framework" section of each individual product's installation documentation to determine if any special actions need to be performed outside of IUF for the `management-nodes-rollout` stage.
+
+1. Use `kubectl` to label all NCN worker nodes but one with `iuf-prevent-rollout=true` to ensure `management-nodes-rollout` only rebuilds a single NCN worker node. This node is referred to as the "canary node" in the remainder of
```

Comment:
> I noticed the bootprep images build Storage and Master node images. Do we need to prevent those from being upgraded too?

Storage nodes and not handled by IUF at this time.  Master nodes are (except for m001), but you have to explicitly target them by specifying the role:subrole.  I'll make this clearer.

Reply:
Fixed in next update.